idaes/core/util/utility_minimization.py

```diff
@@ -0,0 +1,621 @@
+##############################################################################
+"""
+General purpose heat integration block for IDAES models
+"""
+
+# Import Python libraries
+import logging
+
+# Import Plotting and Numpy libraries for composite curves
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Import Pyomo libraries
+from pyomo.environ import (Constraint,
+                           sqrt,
+                           Expression,
+                           value,
+                           Var)
+
+from pyomo.environ import units as pyunits
+
+__author__ = "Alejandro Garciadiego, Alexander Dowling"
+
+def min_utility(blk, heating, cooling, DTmin, eps=1e-6, DG_units=pyunits.Mwatt):
+    """
+    Function for Duran-Grossmann for minimization of utilities
+    This function adds variables and constraints to the flowsheet to minimize utilities
+    Allows the calculation of reactors
+
+    Args:
+        blk: flowsheet
+        heating: Equipment that requires Heat
+        cooling: Equipment from wich heat is being removed
+        DTmin: HRAT (Heat Recovery Approximation Temperature)
+        eps: Epsilon for smoothing operation
+
+    Returns:
+        Constraint and variable object representing the calculation for hot utility and cold utility
+    """
+    epsT = 1
+
+    # Generate lists out of strings from Args
+    exchanger_list = heating + cooling
```

Comment:
It would be good to have some argument verification, but it doesn't need to hold up the PR.

Reply:
I also changed the hardcoded units to use the ones from pyomo